llvm/lib/SYCLLowerIR/CleanupSYCLMetadata.cpp

```diff
@@ -65,7 +64,15 @@ PreservedAnalyses CleanupSYCLMetadataPass::run(Module &M,
   SmallVector<StringRef, 2> ModuleMDToRemove = {"sycl_aspects",
                                                 "sycl_types_that_use_aspects"};
   for (const auto &MD : ModuleMDToRemove)
-    cleanupSYCLCompilerMetadata(M, MD);
+    cleanupSYCLCompilerModuleMetadata(M, MD);
+
+  // Cleanup no longer needed function metadata.
+  SmallVector<StringRef, 1> FunctionMDToRemove = {"srcloc"};
+  for (auto &F : M) {
+    for (const auto &MD : FunctionMDToRemove)
+      if (F.getMetadata(MD))
+        F.setMetadata(MD, nullptr);
```

Comment:
Going further. Having a vector for just one element looks strange.
```suggestion
  for (auto &F : M) {
    if (F.getMetadata("srcloc"))
      F.setMetadata("srcloc", nullptr);
```

Reply:
Applied